presto-native-execution/presto_cpp/main/connectors/arrow_flight/auth/Authenticator.h

```diff
@@ -0,0 +1,85 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include <arrow/flight/api.h>
```

Comment:
We should avoid exposing this arrow header as well.
Can we use PIMPL here as well?

Reply:
No need of PIMPL. Forward declare and make API
```
virtual void authenticateClient(
      arrow::flight::FlightClient* client,
      const velox::config::ConfigBase* sessionProperties,
      arrow::flight::AddCallHeaders* headerWriter) = 0;
```